<a href="https://colab.research.google.com/github/arpitmathur16/Project_imdb/blob/main/Copy_of_project_imdb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [2]:
print(pd.__version__)

2.2.2


In [3]:
df = pd.read_csv("/content/movies.csv")

In [4]:
df

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
0,tt0068646,The Godfather,1972,9.2,R,175 min,"Crime, Drama",1798749,"134,966,411",nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87 min,"Animation, Adventure, Comedy","1,220","134,966,411","nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
2,tt7657566,Death on the Nile,2022,6.3,PG-13,127 min,"Crime, Drama, Mystery","121,063","134,966,411",nm0000110,Kenneth Branagh,"nm5290643,nm0000906,nm0000110,nm1258970","Tom Bateman,Annette Bening,Kenneth Branagh,Rus...","While on vacation on the Nile, Hercule Poirot ..."
3,tt8115900,The Bad Guys,2022,6.9,PG,100 min,"Animation, Adventure, Comedy","23,090","134,966,411",nm2010048,Pierre Perifel,"nm0005377,nm0549505,nm5377144,nm0732497","Sam Rockwell,Marc Maron,Awkwafina,Craig Robinson",Several reformed yet misunderstood criminal an...
4,tt10809742,Collision,2022,3.8,TV-MA,99 min,"Crime, Drama, Thriller",656,"134,966,411",nm1926494,Fabien Martorell,"nm1339181,nm0456810,nm12651040,nm7357127","Tessa Jubber,Langley Kirkwood,Zoey Sneedon,Bon...",Freedom always comes at a price.
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
19803,tt0459423,Kahaani Gudiya Ki...: True Story of a Woman,2008,6.2,Not Rated,112 min,"Biography, Drama",41,"44,435",nm1909606,Prabhakar Shukla,"nm0244890,nm0952232,nm1056425,nm6389513","Divya Dutta,Arif Zakaria,Rajpal Naurang Yadav,...","A tragic saga about a woman named Gudiya, whos..."
19804,tt0119621,Mark Twain's America in,1998,6.5,G,52 min,"Documentary, Biography",48,"2,281,741",nm0522844,Stephen Low,"nm0000843,nm0640297,nm4700843,nm1838044","Anne Bancroft,Dennis O'Connor,Kim Parr,LeeAnn ...",Interweaves Twain's life and times with the li...
19805,tt0168950,Let It Come Down: The Life of Paul Bowles,1998,6.9,Not Rated,75 min,"Documentary, Biography",125,"9,188",nm0047028,Jennifer Baichwal,"nm0101247,nm0123221,nm0155822,nm0159383","Paul Bowles,William S. Burroughs,Cherifa,Moham...",Add a Plot
19806,tt0267563,Hybrid,2000,7.4,Not Rated,92 min,"Documentary, Biography",146,"159,403","/name/nm0566144/,/name/nm1084921/","Monteith McCollum,Milford Beeghly",nm1084921,Milford Beeghly,The life story of Milford Beeghly: Iowa farmer...


Q1 How do we load the dataset and see the firest 5 rows


In [5]:
df.head()

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
0,tt0068646,The Godfather,1972,9.2,R,175 min,"Crime, Drama",1798749,"134,966,411",nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87 min,"Animation, Adventure, Comedy","1,220","134,966,411","nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
2,tt7657566,Death on the Nile,2022,6.3,PG-13,127 min,"Crime, Drama, Mystery","121,063","134,966,411",nm0000110,Kenneth Branagh,"nm5290643,nm0000906,nm0000110,nm1258970","Tom Bateman,Annette Bening,Kenneth Branagh,Rus...","While on vacation on the Nile, Hercule Poirot ..."
3,tt8115900,The Bad Guys,2022,6.9,PG,100 min,"Animation, Adventure, Comedy","23,090","134,966,411",nm2010048,Pierre Perifel,"nm0005377,nm0549505,nm5377144,nm0732497","Sam Rockwell,Marc Maron,Awkwafina,Craig Robinson",Several reformed yet misunderstood criminal an...
4,tt10809742,Collision,2022,3.8,TV-MA,99 min,"Crime, Drama, Thriller",656,"134,966,411",nm1926494,Fabien Martorell,"nm1339181,nm0456810,nm12651040,nm7357127","Tessa Jubber,Langley Kirkwood,Zoey Sneedon,Bon...",Freedom always comes at a price.


In [6]:
df.tail()

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
19803,tt0459423,Kahaani Gudiya Ki...: True Story of a Woman,2008,6.2,Not Rated,112 min,"Biography, Drama",41,"44,435",nm1909606,Prabhakar Shukla,"nm0244890,nm0952232,nm1056425,nm6389513","Divya Dutta,Arif Zakaria,Rajpal Naurang Yadav,...","A tragic saga about a woman named Gudiya, whos..."
19804,tt0119621,Mark Twain's America in,1998,6.5,G,52 min,"Documentary, Biography",48,"2,281,741",nm0522844,Stephen Low,"nm0000843,nm0640297,nm4700843,nm1838044","Anne Bancroft,Dennis O'Connor,Kim Parr,LeeAnn ...",Interweaves Twain's life and times with the li...
19805,tt0168950,Let It Come Down: The Life of Paul Bowles,1998,6.9,Not Rated,75 min,"Documentary, Biography",125,"9,188",nm0047028,Jennifer Baichwal,"nm0101247,nm0123221,nm0155822,nm0159383","Paul Bowles,William S. Burroughs,Cherifa,Moham...",Add a Plot
19806,tt0267563,Hybrid,2000,7.4,Not Rated,92 min,"Documentary, Biography",146,"159,403","/name/nm0566144/,/name/nm1084921/","Monteith McCollum,Milford Beeghly",nm1084921,Milford Beeghly,The life story of Milford Beeghly: Iowa farmer...
19807,tt0116481,Hands on a Hardbody: The Documentary,1997,7.6,PG,98 min,"Documentary, Comedy, Drama",2167,"492,876",nm0082835,S.R. Bindler,"nm0021057,nm1828145,nm0185044,nm0190040","Sid Allen,Brent Baskin,Greg Cox,J.C. Crum",Twenty-four contestants compete in an enduranc...


In [7]:
df.sample(5)

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
3533,tt7671414,Thriller,2018,3.6,TV-MA,87 min,"Crime, Drama, Horror","1,221","14,688,078",nm1371010,Dallas Jackson,"nm6973777,nm8374329,nm5713771,nm1209950","Jessica Allain,Luke Tennie,Mitchell Edwards,Pa...",A childhood prank comes back to haunt a clique...
3860,tt0273045,2001: Do Hazaar Ek,1998,3.8,Not Rated,149 min,"Crime, Mystery, Thriller",130,"1,327,740",nm0802692,Raj N. Sippy,"nm0006763,nm0438092,nm0007102,nm0066073","Jackie Shroff,Dimple Kapadia,Tabu,Rajat Bedi",Following the brutal slaying of a call-girl na...
5222,tt2263942,Massacre Gun,1967,7.0,Not Rated,89 min,"Action, Crime",490,"408,966",nm0367915,Yasuharu Hasebe,"nm0794425,nm0632967,nm0945516,nm0645568","Jô Shishido,Hideaki Nitani,Yôko Yamamoto,Jirô ...",Kuroda (Jô Shishido) is a mob hitman who turns...
2034,tt0068675,Hannie Caulder,1971,6.3,R,85 min,"Crime, Drama, Western",3431,"459,650",nm0447944,Burt Kennedy,"nm0000079,nm0191685,nm0000308,nm0001181","Raquel Welch,Robert Culp,Ernest Borgnine,Jack ...","After she is raped and her husband murdered, a..."
10985,tt2882912,Bratz Girlz Really Rock,2008,5.9,TV-G,78 min,Animation,100,"52,339",nm0268741,David Mucci Fassett,"nm1868320,nm2724682,nm0222175,nm0223740","Ashleigh Ball,Dorla Bell,Trevor Devall,Jessica...",When the Bratz are invited to an exclusive fas...


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19808 entries, 0 to 19807
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              19808 non-null  object 
 1   name            19808 non-null  object 
 2   year            19808 non-null  int64  
 3   rating          19808 non-null  float64
 4   certificate     19808 non-null  object 
 5   duration        19808 non-null  object 
 6   genre           19808 non-null  object 
 7   votes           19808 non-null  object 
 8   gross_income    19808 non-null  object 
 9   directors_id    19808 non-null  object 
 10  directors_name  19808 non-null  object 
 11  stars_id        19808 non-null  object 
 12  stars_name      19808 non-null  object 
 13  description     19808 non-null  object 
dtypes: float64(1), int64(1), object(12)
memory usage: 2.1+ MB


In [9]:
df.describe()

,year,rating
count,19808.000000,19808.000000
mean,1995.546597,6.041973
std,22.801725,1.206831
min,1894.000000,1.100000
25%,1983.000000,5.300000
50%,2003.000000,6.200000
75%,2014.000000,6.900000
max,2022.000000,9.700000


In [10]:
df.duplicated().sum()

np.int64(0)

In [11]:
df.isnull().sum()  # we can also use the isna() in case of isnull() both are same

,0
id,0
name,0
year,0
rating,0
certificate,0
duration,0
genre,0
votes,0
gross_income,0
directors_id,0


In [12]:
# NULL - None
# NA - Not Aviable
# NaN - Not a Number

In [13]:
df["year"]

,year
0,1972
1,2022
2,2022
3,2022
4,2022
...,...
19803,2008
19804,1998
19805,1998
19806,2000


Q3 - How do we clean the duratrion column ? (e.g., '175 min' to 175)

In [14]:
df["duration"] = df["duration"].str.replace("min", "")

In [15]:
df["duration"] = df["duration"].str.replace(",", "")

In [16]:
df["duration"] = df["duration"].astype(int)

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19808 entries, 0 to 19807
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              19808 non-null  object 
 1   name            19808 non-null  object 
 2   year            19808 non-null  int64  
 3   rating          19808 non-null  float64
 4   certificate     19808 non-null  object 
 5   duration        19808 non-null  int64  
 6   genre           19808 non-null  object 
 7   votes           19808 non-null  object 
 8   gross_income    19808 non-null  object 
 9   directors_id    19808 non-null  object 
 10  directors_name  19808 non-null  object 
 11  stars_id        19808 non-null  object 
 12  stars_name      19808 non-null  object 
 13  description     19808 non-null  object 
dtypes: float64(1), int64(2), object(11)
memory usage: 2.1+ MB


In [18]:
df["duration"].mean()

np.float64(101.77796849757674)

In [19]:
df["votes"] = df["votes"].str.replace(",", "")

In [20]:
df["votes"] = df["votes"].astype(int)

In [21]:
df["votes"]

,votes
0,1798749
1,1220
2,121063
3,23090
4,656
...,...
19803,41
19804,48
19805,125
19806,146


In [22]:
df["gross_income"] = df["gross_income"].str.replace(",", "")

In [23]:
df["gross_income"] = df["gross_income"].str.replace("$", "")

In [24]:
df["gross_income"] = df["gross_income"].str.replace("M", "")

In [25]:
df["gross_income"] = df["gross_income"].str.replace(".", "")

In [26]:
df["gross_income"] = df["gross_income"].astype(float)

In [27]:
df["gross_income"]

,gross_income
0,134966411.0
1,134966411.0
2,134966411.0
3,134966411.0
4,134966411.0
...,...
19803,44435.0
19804,2281741.0
19805,9188.0
19806,159403.0


Another way to do it

df['']

In [28]:
df.isnull().sum()

,0
id,0
name,0
year,0
rating,0
certificate,0
duration,0
genre,0
votes,0
gross_income,0
directors_id,0



How do we handle missing value


In [29]:
df.describe()

,year,rating,duration,votes,gross_income
count,19808.000000,19808.000000,19808.000000,1.980800e+04,1.980800e+04
mean,1995.546597,6.041973,101.777968,3.694303e+04,1.682198e+07
std,22.801725,1.206831,43.178721,1.129755e+05,4.642034e+07
min,1894.000000,1.100000,1.000000,5.000000e+00,0.000000e+00
25%,1983.000000,5.300000,90.000000,1.058000e+03,1.196010e+05
50%,2003.000000,6.200000,98.000000,4.233000e+03,1.158192e+06
75%,2014.000000,6.900000,111.000000,2.045350e+04,1.243278e+07
max,2022.000000,9.700000,5220.000000,2.574832e+06,9.366622e+08


In [30]:
df[(df['rating'] > 6.9) & (df["votes"] < 2058.3)]

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87,"Animation, Adventure, Comedy",1220,134966411.0,"nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
319,tt20420316,Gladbeck: The Hostage Crisis,2022,7.5,TV-MA,91,"Documentary, Crime",1317,17266971.0,nm0374657,Volker Heise,"nm3115479,nm3116579","Dieter Degowski,Hans-Jürgen Rösner","In August 1988, two armed bank robbers keep Ge..."
2386,tt0021673,Blonde Crazy,1931,7.1,TV-G,79,"Comedy, Crime, Drama",1967,12282994.0,nm0215877,Roy Del Ruth,"nm0000010,nm0000951,nm0129894,nm0290271","James Cagney,Joan Blondell,Louis Calhern,Noel ...",The adventures of an egoistic con man and his ...
2718,tt0073626,Revolver,1973,7.0,R,111,"Crime, Drama, Thriller",2038,1631839.0,nm0813177,Sergio Sollima,"nm0001657,nm0601377,nm0685632,nm0068941","Oliver Reed,Fabio Testi,Paola Pitagora,Agostin...",An Italian prison warden is forced to exchange...
2820,tt14035242,A Cop Movie,2021,7.0,R,107,"Documentary, Action, Crime",1725,9881286.0,nm2091911,Alonso Ruizpalacios,"nm2861211,nm3028204,nm2415604,nm2660250","Leonardo Alonso,Raúl Briones,Mónica Del Carmen...",Two professional actors undergo an immersive p...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
19799,tt1486548,Ahead of Time: The Extraordinary Journey of Ru...,2009,7.9,Not Rated,73,"Documentary, Biography",77,33024.0,nm0725053,Robert Richman,"nm0344295,nm4271124,nm0811750,nm4270360","Ruth Gruber,Louisa Herz Leopold,Dava Sobel,Lee...",A documentary on the remarkable life of Ruth G...
19800,tt0044917,Amazing Monsieur Fabre,1951,7.0,Approved,90,"Biography, Comedy, Drama",64,33024.0,nm0224543,Henri Diamant-Berger,"nm0294382,nm0479626,nm0709536,nm0845969","Pierre Fresnay,Elina Labourdette,André Randall...",Pierre Fresnay plays the title role of a dedic...
19801,tt1334479,"Yoo-Hoo, Mrs. Goldberg",2009,7.4,Unrated,92,"Documentary, Biography",236,1133662.0,nm0447476,Aviva Kempner,"nm0073764,nm3571106,nm1675671,nm0046029","Gertrude Berg,Lewis Berg,Sara Chase,Ruth Bader...","The story of the actress, writer, and broadcas..."
19802,tt1518255,Raw Faith,2010,8.3,Not Rated,94,"Documentary, Biography, Drama",71,1486.0,"/name/nm0927286/,/name/nm3628867/","Wm. Peter Wiedensmith,Marilyn Sewell",nm3628867,Marilyn Sewell,This surprisingly open and revealing documenta...


In [31]:
df[df["votes"] > 1058]

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
0,tt0068646,The Godfather,1972,9.2,R,175,"Crime, Drama",1798749,134966411.0,nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87,"Animation, Adventure, Comedy",1220,134966411.0,"nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
2,tt7657566,Death on the Nile,2022,6.3,PG-13,127,"Crime, Drama, Mystery",121063,134966411.0,nm0000110,Kenneth Branagh,"nm5290643,nm0000906,nm0000110,nm1258970","Tom Bateman,Annette Bening,Kenneth Branagh,Rus...","While on vacation on the Nile, Hercule Poirot ..."
3,tt8115900,The Bad Guys,2022,6.9,PG,100,"Animation, Adventure, Comedy",23090,134966411.0,nm2010048,Pierre Perifel,"nm0005377,nm0549505,nm5377144,nm0732497","Sam Rockwell,Marc Maron,Awkwafina,Craig Robinson",Several reformed yet misunderstood criminal an...
5,tt0071562,The Godfather Part II,1974,9.0,R,202,"Crime, Drama",1239038,57300000.0,nm0000338,Francis Ford Coppola,"nm0000199,nm0000134,nm0000380,nm0000473","Al Pacino,Robert De Niro,Robert Duvall,Diane K...",The early life and career of Vito Corleone in ...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
19737,tt1972646,Eames: The Architect & The Painter,2011,7.1,Not Rated,85,"Documentary, Biography",1096,147591.0,"nm2912462,nm0422080","Jason Cohn,Bill Jersey","nm0247177,nm0247194,nm0290556,nm0649223","Charles Eames,Ray Eames,James Franco,Jeannine ...",The relationship between Charles Eames and his...
19740,tt2174896,To Be Takei,2014,6.9,Not Rated,94,"Documentary, Biography",1533,51481.0,nm1464376,Jennifer M. Kroot,"nm0001786,nm2811871,nm0000479,nm0000638","George Takei,Brad Takei,Walter Koenig,William ...",A look at the many roles played by eclectic ac...
19746,tt1839590,Ray Harryhausen: Special Effects Titan,2011,7.5,Not Rated,90,"Documentary, Biography",1312,166770.0,nm1583697,Gilles Penso,"nm0366063,nm0001392,nm0000416,nm0868219","Ray Harryhausen,Peter Jackson,Terry Gilliam,Gu...",The life and work of the renowned film visual ...
19789,tt0492499,An Unreasonable Man,2006,7.9,Not Rated,122,"Documentary, Biography",1728,175237.0,"nm0544055,nm0804887","Henriette Mantel,Steve Skrovan","nm0618873,nm0118059,nm0957016,nm1736409","Ralph Nader,Pat Buchanan,Howard Zinn,Eric Alte...",A look at the career of consumer advocate Ralp...


In [32]:
df[df["votes"] > 20458.3]

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
0,tt0068646,The Godfather,1972,9.2,R,175,"Crime, Drama",1798749,134966411.0,nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
2,tt7657566,Death on the Nile,2022,6.3,PG-13,127,"Crime, Drama, Mystery",121063,134966411.0,nm0000110,Kenneth Branagh,"nm5290643,nm0000906,nm0000110,nm1258970","Tom Bateman,Annette Bening,Kenneth Branagh,Rus...","While on vacation on the Nile, Hercule Poirot ..."
3,tt8115900,The Bad Guys,2022,6.9,PG,100,"Animation, Adventure, Comedy",23090,134966411.0,nm2010048,Pierre Perifel,"nm0005377,nm0549505,nm5377144,nm0732497","Sam Rockwell,Marc Maron,Awkwafina,Craig Robinson",Several reformed yet misunderstood criminal an...
5,tt0071562,The Godfather Part II,1974,9.0,R,202,"Crime, Drama",1239038,57300000.0,nm0000338,Francis Ford Coppola,"nm0000199,nm0000134,nm0000380,nm0000473","Al Pacino,Robert De Niro,Robert Duvall,Diane K...",The early life and career of Vito Corleone in ...
6,tt4998632,Ambulance,2022,6.1,R,136,"Action, Crime, Drama",49505,57300000.0,nm0000881,Michael Bay,"nm0350453,nm5584344,nm2555462,nm0226813","Jake Gyllenhaal,Yahya Abdul-Mateen II,Eiza Gon...",Two robbers steal an ambulance after their hei...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
19374,tt2652092,The Good Lie,2014,7.4,PG-13,110,"Biography, Drama",28589,2716989.0,nm0265852,Philippe Falardeau,"nm0000702,nm0643668,nm1563346,nm2897500","Reese Witherspoon,Arnold Oceng,Ger Duany,Emman...","A group of Sudanese refugees, given the chance..."
19385,tt7485048,Super 30,2019,7.9,Not Rated,154,"Biography, Drama",30305,2269878.0,nm2134474,Vikas Bahl,"nm0004335,nm7021807,nm5446189,nm0768297","Hrithik Roshan,Mrunal Thakur,Nandish Singh San...",Based on the life of Patna-based mathematician...
19390,tt1035736,Coco Before Chanel,2009,6.6,PG-13,105,"Biography, Drama",40496,6109075.0,nm0284774,Anne Fontaine,"nm0851582,nm0688143,nm0005273,nm0002100","Audrey Tautou,Benoît Poelvoorde,Alessandro Niv...",The story of Coco Chanel's rise from obscure b...
19542,tt6910020,Nothing to Lose,2018,2.3,PG,134,"Biography, Drama",23028,1003497.0,nm0042768,Alexandre Avancini,"nm0327404,nm2780552,nm0332283,nm0896942","Petrônio Gontijo,Day Mesquita,Beth Goulart,Dal...",Self-financed fictional biopic of Brazilian bi...


Q8. Find the "overhyped" movies (Low Rating, High Gross)

In [33]:
df.describe()

,year,rating,duration,votes,gross_income
count,19808.000000,19808.000000,19808.000000,1.980800e+04,1.980800e+04
mean,1995.546597,6.041973,101.777968,3.694303e+04,1.682198e+07
std,22.801725,1.206831,43.178721,1.129755e+05,4.642034e+07
min,1894.000000,1.100000,1.000000,5.000000e+00,0.000000e+00
25%,1983.000000,5.300000,90.000000,1.058000e+03,1.196010e+05
50%,2003.000000,6.200000,98.000000,4.233000e+03,1.158192e+06
75%,2014.000000,6.900000,111.000000,2.045350e+04,1.243278e+07
max,2022.000000,9.700000,5220.000000,2.574832e+06,9.366622e+08


In [34]:
df[(df['rating'] < 5.3) & (df["gross_income"] > 12432780)]

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
4,tt10809742,Collision,2022,3.8,TV-MA,99,"Crime, Drama, Thriller",656,134966411.0,nm1926494,Fabien Martorell,"nm1339181,nm0456810,nm12651040,nm7357127","Tessa Jubber,Langley Kirkwood,Zoey Sneedon,Bon...",Freedom always comes at a price.
18,tt11656220,Midnight in the Switchgrass,2021,4.4,R,99,"Action, Crime, Thriller",6827,107928762.0,nm0256542,Randall Emmett,"nm0386472,nm1083271,nm0000246,nm0001305","Emile Hirsch,Megan Fox,Bruce Willis,Lukas Haas",An FBI agent and Florida State officer team up...
37,tt6749318,Speed Kills,2018,4.2,R,102,"Action, Crime, Drama",3381,130742922.0,nm9928566,Jodi Scurfield,"nm0000237,nm0935395,nm0261170,nm0922995","John Travolta,Katheryn Winnick,Jennifer Esposi...",Speedboat racing champion and multimillionaire...
40,tt5433138,F9: The Fast Saga,2021,5.2,PG-13,143,"Action, Crime, Thriller",125648,173202780.0,nm0510912,Justin Lin,"nm0004874,nm0735442,nm0108287,nm0879085","Vin Diesel,Michelle Rodriguez,Jordana Brewster...",Dom and the crew must take on an international...
47,tt1551614,Blowback,2022,4.4,R,93,"Action, Crime, Thriller",199,74283625.0,nm0847749,Tibor Takács,"nm1330276,nm1544217,nm6029604,nm0541909","Randy Couture,Cam Gigandet,Michele Plaia,Louis...","After planning a perfect heist, Nick is ready ..."
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
18575,tt1389139,When the Bough Breaks,2016,5.1,PG-13,107,"Drama, Thriller",6974,29703843.0,nm0143984,Jon Cassar,"nm0004820,nm0356021,nm0539082,nm0931324","Morris Chestnut,Regina Hall,Romany Malco,Micha...",A surrogate mom for a couple becomes dangerous...
18615,tt8170298,John Henry,2020,3.6,R,91,"Drama, Thriller",2368,21082165.0,nm3112158,Will Forbes,"nm0187719,nm4175452,nm0524839,nm0286010","Terry Crews,Jamila Velazquez,Ludacris,Ken Foree",When two immigrant kids on the run from his fo...
18616,tt6469960,An Imperfect Murder,2017,3.2,R,71,Thriller,1666,15156200.0,nm0864812,James Toback,"nm1092227,nm0000285,nm0001301,nm0131974","Sienna Miller,Alec Baldwin,Charles Grodin,Coll...",An actress's nightmare about killing her ex-bo...
18717,tt1859573,Pretty Obsession,2012,4.2,Not Rated,89,Thriller,193,12974636.0,nm0062341,Michael Baumgarten,"nm4386106,nm4672255,nm2373963,nm2949570","Brittany Nicole Kovler,Lydia Pritchett,Danny P...",A pretty eighteen year-old puts herself and he...


 Q9. What are the unique certificates in the dataset?

In [35]:
df["certificate"].unique()

array(['R', 'PG', 'PG-13', 'TV-MA', 'Not Rated', 'Approved', 'TV-14',
       'NC-17', 'Unrated', 'G', 'Passed', 'GP', 'X', 'M/PG', 'M', 'TV-PG',
       'TV-G', '12', 'TV-13', 'TV-Y7-FV', 'MA-17', 'AO', '18', 'R-15',
       'R-12', 'R-18', 'PG-12', '6+', 'TV-Y7', 'TV-Y', 'MA-13'],
      dtype=object)

How many Unique values are there

In [36]:
df["certificate"].nunique()

31

Q10. How many movies were released each year?


In [37]:
df['year'].value_counts().sort_index(ascending=True)

,count
year,
1894,1
1898,1
1902,2
1903,1
1913,3
...,...
2018,736
2019,657
2020,467


Another way to find the movie releases in a year

In [38]:
df.groupby("year")["id"].count().sort_index(ascending=False)

,id
year,
2022,174
2021,470
2020,467
2019,657
2018,736
...,...
1913,3
1903,1
1902,2


In [43]:
x = df["genre"].str.split(",").explode()

x = x.str.strip()

In [48]:
x.value_counts().sort_values(ascending= False).head(10)

,count
genre,
Drama,9655
Action,5290
Crime,5244
Comedy,4641
Thriller,4231
Adventure,3313
Horror,3143
Mystery,2211
Romance,1815


Q12. Which directors have the most movies ?

In [50]:
df.head(5)

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
0,tt0068646,The Godfather,1972,9.2,R,175,"Crime, Drama",1798749,134966411.0,nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87,"Animation, Adventure, Comedy",1220,134966411.0,"nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
2,tt7657566,Death on the Nile,2022,6.3,PG-13,127,"Crime, Drama, Mystery",121063,134966411.0,nm0000110,Kenneth Branagh,"nm5290643,nm0000906,nm0000110,nm1258970","Tom Bateman,Annette Bening,Kenneth Branagh,Rus...","While on vacation on the Nile, Hercule Poirot ..."
3,tt8115900,The Bad Guys,2022,6.9,PG,100,"Animation, Adventure, Comedy",23090,134966411.0,nm2010048,Pierre Perifel,"nm0005377,nm0549505,nm5377144,nm0732497","Sam Rockwell,Marc Maron,Awkwafina,Craig Robinson",Several reformed yet misunderstood criminal an...
4,tt10809742,Collision,2022,3.8,TV-MA,99,"Crime, Drama, Thriller",656,134966411.0,nm1926494,Fabien Martorell,"nm1339181,nm0456810,nm12651040,nm7357127","Tessa Jubber,Langley Kirkwood,Zoey Sneedon,Bon...",Freedom always comes at a price.


In [51]:
y = df["directors_name"].str.split(",").explode()

y = y.str.strip()

In [52]:
y.value_counts().sort_values(ascending= False).head(10)

,count
directors_name,
Clint Eastwood,38
Alfred Hitchcock,38
Henry Hathaway,31
John Ford,31
Raoul Walsh,31
Steven Spielberg,30
John Huston,29
Jesús Franco,28
Martin Scorsese,26


Q13. Do longer movies get bettere ratings?

In [54]:
df[["duration","rating"]].corr()

,duration,rating
duration,1.000000,0.130112
rating,0.130112,1.000000


In [55]:
df[["votes", "gross_income"]].corr()

,votes,gross_income
votes,1.000000,0.632683
gross_income,0.632683,1.000000


In [56]:
df[["certificate", "gross_income"]].corr()

ValueError: could not convert string to float: 'R'

Q14. Which genre have the highest rating ?


In [59]:
df

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
0,tt0068646,The Godfather,1972,9.2,R,175,"Crime, Drama",1798749,134966411.0,nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87,"Animation, Adventure, Comedy",1220,134966411.0,"nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
2,tt7657566,Death on the Nile,2022,6.3,PG-13,127,"Crime, Drama, Mystery",121063,134966411.0,nm0000110,Kenneth Branagh,"nm5290643,nm0000906,nm0000110,nm1258970","Tom Bateman,Annette Bening,Kenneth Branagh,Rus...","While on vacation on the Nile, Hercule Poirot ..."
3,tt8115900,The Bad Guys,2022,6.9,PG,100,"Animation, Adventure, Comedy",23090,134966411.0,nm2010048,Pierre Perifel,"nm0005377,nm0549505,nm5377144,nm0732497","Sam Rockwell,Marc Maron,Awkwafina,Craig Robinson",Several reformed yet misunderstood criminal an...
4,tt10809742,Collision,2022,3.8,TV-MA,99,"Crime, Drama, Thriller",656,134966411.0,nm1926494,Fabien Martorell,"nm1339181,nm0456810,nm12651040,nm7357127","Tessa Jubber,Langley Kirkwood,Zoey Sneedon,Bon...",Freedom always comes at a price.
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
19803,tt0459423,Kahaani Gudiya Ki...: True Story of a Woman,2008,6.2,Not Rated,112,"Biography, Drama",41,44435.0,nm1909606,Prabhakar Shukla,"nm0244890,nm0952232,nm1056425,nm6389513","Divya Dutta,Arif Zakaria,Rajpal Naurang Yadav,...","A tragic saga about a woman named Gudiya, whos..."
19804,tt0119621,Mark Twain's America in,1998,6.5,G,52,"Documentary, Biography",48,2281741.0,nm0522844,Stephen Low,"nm0000843,nm0640297,nm4700843,nm1838044","Anne Bancroft,Dennis O'Connor,Kim Parr,LeeAnn ...",Interweaves Twain's life and times with the li...
19805,tt0168950,Let It Come Down: The Life of Paul Bowles,1998,6.9,Not Rated,75,"Documentary, Biography",125,9188.0,nm0047028,Jennifer Baichwal,"nm0101247,nm0123221,nm0155822,nm0159383","Paul Bowles,William S. Burroughs,Cherifa,Moham...",Add a Plot
19806,tt0267563,Hybrid,2000,7.4,Not Rated,92,"Documentary, Biography",146,159403.0,"/name/nm0566144/,/name/nm1084921/","Monteith McCollum,Milford Beeghly",nm1084921,Milford Beeghly,The life story of Milford Beeghly: Iowa farmer...


In [69]:
df.assign(genre = df["genre"].str.split(",")).explode("genre")

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
0,tt0068646,The Godfather,1972,9.2,R,175,Crime,1798749,134966411.0,nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
0,tt0068646,The Godfather,1972,9.2,R,175,Drama,1798749,134966411.0,nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87,Animation,1220,134966411.0,"nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87,Adventure,1220,134966411.0,"nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87,Comedy,1220,134966411.0,"nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
19806,tt0267563,Hybrid,2000,7.4,Not Rated,92,Documentary,146,159403.0,"/name/nm0566144/,/name/nm1084921/","Monteith McCollum,Milford Beeghly",nm1084921,Milford Beeghly,The life story of Milford Beeghly: Iowa farmer...
19806,tt0267563,Hybrid,2000,7.4,Not Rated,92,Biography,146,159403.0,"/name/nm0566144/,/name/nm1084921/","Monteith McCollum,Milford Beeghly",nm1084921,Milford Beeghly,The life story of Milford Beeghly: Iowa farmer...
19807,tt0116481,Hands on a Hardbody: The Documentary,1997,7.6,PG,98,Documentary,2167,492876.0,nm0082835,S.R. Bindler,"nm0021057,nm1828145,nm0185044,nm0190040","Sid Allen,Brent Baskin,Greg Cox,J.C. Crum",Twenty-four contestants compete in an enduranc...
19807,tt0116481,Hands on a Hardbody: The Documentary,1997,7.6,PG,98,Comedy,2167,492876.0,nm0082835,S.R. Bindler,"nm0021057,nm1828145,nm0185044,nm0190040","Sid Allen,Brent Baskin,Greg Cox,J.C. Crum",Twenty-four contestants compete in an enduranc...


In [66]:
df.head(2)

,id,name,year,rating,certificate,duration,genre,votes,gross_income,directors_id,directors_name,stars_id,stars_name,description
0,tt0068646,The Godfather,1972,9.2,R,175,"Crime, Drama",1798749,134966411.0,nm0000338,Francis Ford Coppola,"nm0000008,nm0000199,nm0001001,nm0000473","Marlon Brando,Al Pacino,James Caan,Diane Keaton",The aging patriarch of an organized crime dyna...
1,tt5113044,Minions: The Rise of Gru,2022,7.3,PG,87,"Animation, Adventure, Comedy",1220,134966411.0,"nm0049633,nm1556070,nm3646390","Kyle Balda,Brad Ableson,Jonathan del Val","nm0136797,nm1853544,nm0000273,nm0378245","Steve Carell,Pierre Coffin,Alan Arkin,Taraji P...",The untold story of one twelve-year-old's drea...


In [70]:
df2.groupby("genre")["rating"].mean().sort_values(ascending = False).head(5)

,rating
genre,
Documentary,7.187100
Short,7.117143
Reality-TV,7.100000
Biography,7.090317
Short,6.997674
